pkg/internalapis/featurestates/featurestates.go

```
@@ -0,0 +1,438 @@
+		log.Warnf("configMapAdded: unrecognized object %+v", obj)
+		return
+	}
+	if fssConfigMap.Name == supervisorFeatureStatConfigMapName &&
+		fssConfigMap.Namespace == supervisorFeatureStateConfigMapNamespace {
+		var err error
+		var featureStates []featurestatesv1alpha1.FeatureState
+		for feature, state := range fssConfigMap.Data {
+			var featureState featurestatesv1alpha1.FeatureState
+			featureState.Name = feature
+			featureState.Enabled, err = strconv.ParseBool(state)
+			if err != nil {
+				log.Errorf("failed to parse feature state value: %v for feature: %q", state, feature)
+			}
+			featureStates = append(featureStates, featureState)
+		}
+		pendingCRUpdatesObj.enqueueFeatureStateUpdatesForAllWorkloadNamespaces(ctx, featureStates)
+	}
+}
+
+// configMapUpdated is called when configmap is updated
+func configMapUpdated(oldObj, newObj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	newfssConfigMap, ok := newObj.(*v1.ConfigMap)
+	if newfssConfigMap == nil || !ok {
+		log.Warnf("configMapUpdated: unrecognized old object %+v", newObj)
+		return
+	}
+	oldfssConfigMap, ok := oldObj.(*v1.ConfigMap)
+	if oldfssConfigMap == nil || !ok {
+		log.Warnf("configMapUpdated: unrecognized new object %+v", newObj)
+		return
+	}
+
+	if newfssConfigMap.Name == supervisorFeatureStatConfigMapName &&
+		newfssConfigMap.Namespace == supervisorFeatureStateConfigMapNamespace {
+		if reflect.DeepEqual(newfssConfigMap.Data, oldfssConfigMap.Data) {
+			return
+		}
+		var err error
+		var featureStates []featurestatesv1alpha1.FeatureState
+		for feature, state := range newfssConfigMap.Data {
+			var featureState featurestatesv1alpha1.FeatureState
+			featureState.Name = feature
+			featureState.Enabled, err = strconv.ParseBool(state)
+			if err != nil {
+				log.Errorf("failed to parse feature state value: %v for feature: %q", state, feature)
```

Same comment as below, will this potentially cause any NPE if we ignore errors. Can you set `featureState.Enabled` = `false`

---

If ParseBool fails it returns false with error, so we are good.